src/CMakeLists.txt

```diff
@@ -340,12 +341,10 @@ endif()
 # - external if FLANN_INCLUDE_DIR_HINTS and a valid Flann setup is found
 # ==============================================================================
 if(ALICEVISION_BUILD_SFM)
-  if(NOT DEFINED FLANN_INCLUDE_DIR_HINTS)
-    set(FLANN_INCLUDE_DIR_HINTS ${CMAKE_CURRENT_SOURCE_DIR}/dependencies/flann/src/cpp)
-    set(ALICEVISION_USE_INTERNAL_FLANN ON)
-  endif()
   find_package(Flann QUIET)
   if(NOT FLANN_FOUND OR ALICEVISION_USE_INTERNAL_FLANN)
+    set(FLANN_INCLUDE_DIR_HINTS ${CMAKE_CURRENT_SOURCE_DIR}/dependencies/flann/src/cpp)
```

Comment:
`FLANN_INCLUDE_DIR_HINTS` is now set after `find_package`.
In order to find the internal one we should add another `find_package` after this line.